asserts/findwildcard.go

```diff
@@ -0,0 +1,94 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package asserts
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+/*
+findWildcard invokes foundCb for each regular file matching:
+
+<top>/<descendantWithWildcard[0]>/<descendantWithWildcard[1]...
+
+where each descendantWithWildcard component can contain the * wildcard;
+
+foundCb is invoked with the path of the file relative to top (that means top/
+ is excluded).
+
+Any I/O operation error stops the walking and bottoms out, so a foundCb invocation that returns an error.
+*/
+func findWildcard(top string, descendantWithWildcard []string, foundCb func(relpath string) error) error {
+	return findWildcardDescend(top, top, descendantWithWildcard, foundCb)
+}
+
+func findWildcardDescend(top, current string, descendantWithWildcard []string, foundCb func(relpath string) error) error {
```

Comment:
This looks a lot like a custom implementation of filepath.Glob(). I'm sure I miss something, but what is the advantage over something like this:
```
func findWildcard(top string, descendantWithWildcard []string, foundCb func(relpath string) error) error {
	globP := filepath.Join(top, strings.Join(descendantWithWildcard, "/"))
	matches, err := filepath.Glob(globP)
	if err != nil {
		return err
	}
	for _, match := range matches {
		if err := foundCb(relpath[len(top)+1:]); err != nil {
			return err
		}
	}

	return nil
}
```
(this is of course a bit simplistic, before foundCb() this code would have to do the same checks as the current code is doing in len(descendantWithWildcard.OTOH, the caller could check for IsRegular() too to make it more generic). But it seems like the second part of the code could also use filepath.Glob? If not it would be nice to have a short comment somewhere why this is not the case.